test/functional/helpers.lua

```
@@ -500,9 +500,15 @@ function module.source(code)
 end
 
 function module.set_shell_powershell()
+  local shell = iswin() and 'powershell' or 'pwsh'
+  if not module.eval('executable("'..shell..'")') then
+    error(shell..' is not executable')
+  end
+  module.command('set shell='..shell)
   module.source([[
-    set shell=powershell shellquote=( shellpipe=\| shellredir=> shellxquote=
-    let &shellcmdflag = '-NoLogo -NoProfile -ExecutionPolicy RemoteSigned -Command Remove-Item -Force alias:sleep; Remove-Item -Force alias:cat;'
+    set shellquote= shellpipe=\| shellxquote=
+    let &shellredir = '| Out-File -Encoding UTF8'
+    let &shellcmdflag = '-NoLogo -NoProfile -ExecutionPolicy RemoteSigned -Command Remove-Item -Force alias:sleep 2> $null; Remove-Item -Force alias:cat 2> $null;'
```

Maybe check iswin instead of ignoring errors ?